### PR TITLE
API: Add history tab. Migrate content style variants to BaseElement

### PR DESCRIPTION
**Pull request by wilr**
History tab changes and report.

Blockers:

- [x] https://github.com/silverstripe/silverstripe-versioned/pull/60
- [x] https://github.com/silverstripe/silverstripe-framework/pull/7500
- [x] Peer review 
- [x] Tests for revert button functionality and report view.

Report blockers:

- [x] ~https://github.com/silverstripe/silverstripe-reports/pull/88~
- [x] ~UX review for report view.~

### CODE REVIEW

File: src/Models/BaseElement.php [hunk 1]
```diff
@@ -274,52 +278,46 @@ public function getCMSFields()
...
-        }
-        foreach ($elements as $key => $value) {
-            if ($key == self::class) {
-                unset($elements[$key]);
-                continue;
-            }
-            $elements[$key] = $this->stripNamespacing($value);
-        }
+        $config = GridFieldConfig_RecordViewer::create();
+        $config->removeComponentsByType('SilverStripe\Forms\GridField\GridFieldPageCount');
```
Comment (robbieaverill) [inline src/Models/BaseElement.php]: Minor: it'd be a little more consistent to use `GridFieldPageCount::class` here

**Comment by robbieaverill** [comment]
Updates required:

* Core: don't show GridField header if empty string is given (null renders the GridField name, empty string renders a tr+th with nothing in it)
* Remove sort on ID column
* Hide actions via JS when going into History tab, show again when changing out of it
* Don't show draft records when they were published immediately (e.g. draft ID === live ID) - check CMS page history to see how this is handled there
* Show draft records if saved then revisited later and published with no extra changes

**Comment by robbieaverill** [comment]
A few issues came up while testing this, I'll log them individually once merged.

**Comment by robbieaverill** [comment]
Report removed from this pull request, it will be reintroduced in a separate pull request.

File: README.md [hunk 2]
```diff
@@ -102,7 +102,28 @@ class MyElement extends BaseElement
...
 
-`MyElement` will be rendered into a `MyElement.ss` template with the `ElementHolder.ss` wrapper.
+`MyElement` will be rendered into a `MyElement.ss` template with the `ElementHolder.ss` wrapper. Changing the holder
+template can be done via `YAML` or using a `$controller_template` on your subclass.
+
+```php
+private static $controller_template = 'MyElementHolder';
+```
+
+To customise existing block templates such as `Content` and `Form` templates copy the relevant files from
```
Comment (NightJar) [inline README.md]: >`- templates copy`
>`+ templates, copy`

File: src/Models/ElementContent.php [hunk 3]
```diff
@@ -11,8 +11,7 @@ class ElementContent extends BaseElement
     private static $icon = 'dnadesign/silverstripe-elemental:images/content.svg';
 
     private static $db = [
-        'HTML' => 'HTMLText',
-        'Style' => 'Varchar(255)'
```
Comment (NightJar) [inline src/Models/ElementContent.php]: Why was this removed?

Comment (robbieaverill) [reply src/Models/ElementContent.php]: Moved to base element

**Comment by robbieaverill** [comment]
@dylan I've fixed the docs typo and rebased
